src/portable/platform.rs

```diff
@@ -96,9 +96,14 @@ pub fn optional_docker_check() -> anyhow::Result<bool> {
 }
 
 pub fn is_arm64_hardware() -> bool {
-    process::Native::new("uname", "uname", "uname")
+    process::Native::new("uname", "uname", "arch")
```

Comment:
This prints "arch: Unknown architecture: arm64e" to stderr on x86 macs.  Can we inhibit stderr?  Or maybe we can use `uname -m` and then detect Rosetta explicitly (`$(sysctl -in sysctl.proc_translated) = 1`).